purchases/src/androidTestIntegrationTest/kotlin/com/revenuecat/purchases/offlineentitlements/OfflineEntitlementsWithInitialRequestsCompletedIntegrationTest.kt

```diff
@@ -94,6 +96,29 @@ class OfflineEntitlementsWithInitialRequestsCompletedAndInitialPurchasesIntegrat
         }
     }
 
+    @Test
+    fun entersOfflineEntitlementsModeIfCachedCustomerInfoAndPostingPendingPurchasesReturns500() {
+        ensureBlockFinishes { latch ->
+            Purchases.sharedInstance.forceServerErrors = true
+
+            Purchases.sharedInstance.getCustomerInfoWith(
+                fetchPolicy = CacheFetchPolicy.FETCH_CURRENT,
+                onError = {
+                    fail("Expected success but got error: $it")
+                },
+                onSuccess = { receivedCustomerInfo ->
+                    assertCustomerInfoHasExpectedPurchaseData(receivedCustomerInfo)
+                    latch.countDown()
+                },
+            )
+        }
+    }
+}
+
+@RunWith(AndroidJUnit4::class)
+class OfflineEntitlementsWithInitialRequestsCompletedAndNoInitialPurchasesIntegrationTest :
+    BaseOfflineEntitlementsWithInitialRequestsCompletedIntegrationTest() {
```

Comment:
This is misleading... Here I moved the `doesNotEnterOfflineEntitlementsModeIfCachedCustomerInfoAndCustomerInfoRequestReturns500` test to the `OfflineEntitlementsWithInitialRequestsCompletedAndNoInitialPurchasesIntegrationTest` test class.

This is because, now that we sync pending transactions when getting customer info, we would actually enter offline entitlements in this case of having an initial purchase. The case mentioned above is covered in the new test added above `entersOfflineEntitlementsModeIfCachedCustomerInfoAndPostingPendingPurchasesReturns500`.